tasks/move-granules/index.js

```diff
@@ -331,57 +293,46 @@ async function updateEachCmrFileAccessURLs(
  * @param {Object} event - Lambda function payload
  * @param {Object} event.config - the config object
  * @param {string} event.config.bucket - Bucket name where public/private keys are stored
- * @param {string} event.config.granuleIdExtraction - regex needed to extract granuleId
- *                                                    from filenames
- * @param {Array} event.config.input_granules - an array of granules
+ * @param {Object} event.config.buckets - Buckets config
  * @param {string} event.config.distribution_endpoint - distribution endpoint for the api
  * @param {Object} event.config.collection - collection configuration
  *                     https://nasa.github.io/cumulus/docs/data-cookbooks/setup#collections
  * @param {boolean} [event.config.moveStagedFiles=true] - set to false to skip moving files
  *                                 from staging to final bucket. Mostly useful for testing.
- * @param {Array} event.input - an array of s3 uris
+ * @param {Object} event.input - a granules object containing an array of granules
  * @returns {Promise} returns the promise of an updated event object
  */
 async function moveGranules(event) {
   // we have to post the meta-xml file of all output granules
   // first we check if there is an output file
   const config = get(event, 'config');
-  const sourceBucket = get(config, 'bucket'); // the name of the bucket with staged data.
   const bucketsConfig = new BucketsConfig(get(config, 'buckets'));
-  const granuleIdExtractionRegex = get(config, 'granuleIdExtraction', '(.*)');
-  const inputGranules = get(config, 'input_granules', {});
-  const distEndpoint = get(config, 'distribution_endpoint');
   const moveStagedFiles = get(config, 'moveStagedFiles', true);
-  const collection = config.collection;
 
   const duplicateHandling = duplicateHandlingType(event);
 
-  const inputFileList = get(event, 'input', []);
-
-  // Get list of cmr file objects from the input Array of S3 filenames (in
-  // staging location after processing)
-  const cmrFiles = getCmrFiles(inputFileList, granuleIdExtractionRegex);
-
-  const allGranules = mergeInputFilesWithInputGranules(
-    inputFileList, inputGranules, granuleIdExtractionRegex
-  );
+  const granulesInput = get(event, 'input.granules', []);
```

Comment:
The input schema should ensure that this will be present and it will be an array.

```suggestion
  const granulesInput = event.input.granules;
```

Reply:
👍 2a9af61a4065